### ISSUE TITLE
Reconsidering mesh-external traffic tests

### ISSUE THREAD

**Issue by ronenschafferibm** [istio/istio#5504]
The current situation:
The e2e_pilot tests related to mesh-external traffic that run by Prow and CircleCI are skipped because the `Egress` flag is false. https://github.com/istio/istio/blob/release-0.8/tests/istio.mk#L165

The relevant tests are: `egress_rules`, `egressgateway`, `externalservice`, ` kubernetes_external_name_services`, `routing_to_egress`.
If I understood correctly, that flag is set to false because we don’t want the test to fail when the external service is temporary not available.

As I see it we have 3 options:
1.	Change nothing. Use the original external services.
2.	Deploy and make use of the external services on the test environment, external to the mesh (like app `t`).
3.	Deploy and make use of the external services on a dedicated environment, so it’ll be always up, monitored by the istio team and be used by istio tests exclusively.

Any comments/ideas are welcome.

@rshriram @costin @andra @ldemailly @hklai

**Comment by ldemailly** [comment]
We could use fortio.istio.io or istio.io as external service. We may still have to have some retry or increased timeout at least when connecting from circleci.

**Comment by rshriram** [comment]
Egress is no longer needed in 0.9+. We also don’t need routing to egress as that test is already present (egress gateway tests that use service entry). So the only other thing that needs testing is a service entry with a true mesh external service. Once SNI routing lands, we could simply make a call to google.com and be done with it. Httpbin would have been nice to check for corner cases in http/http2 but I think in the interest of having a test setup that works consistently, we can forgo that.

**Comment by vadimeisenbergibm** [comment]
I propose the following:
1. Deploy httpbin to fortio.istio.io. It is convenient to use httpbin for testing headers, and other HTTP-related features. @ldemailly could you please do it? So we could access fortio.istio.io/httpbin ?
3. Are the IP ranges of fortio.istio.io known?
4. We can still use google.com, until they block us :)
5. I propose that we support all the tests for all the possible options of real mesh-external services. Testing TLS origination, traffic thru egress gateway, TLS with SNI, plain TCP, non-encrypted HTTP, traffic thru Kubernetes external-name services etc. 
6. I propose that we test egress rules as well, until they are removed in 0.9.

**Comment by vadimeisenbergibm** [comment]
@ldemailly Great, let me rewrite the egress tests to use https://fortio.istio.io/debug instead of httpbin.org.

**Comment by vadimeisenbergibm** [comment]
@rshriram My thought was that the availability of istio.io is at least in our hands, including the firewall issues. google.com can detect at some point that there is a lot traffic from our prow/circle_ci and start blocking it. httpbin.org started to fail lately, so we cannot use it anymore. We have to do something ASAP.

Currently we test on httpbin.org (for headers), google.com and wikipedia.org. I propose first to switch httpbin.org to fortio.istio.io/debug. Then if any google.com or wikipedia.org will give us any problems, we will switch their tests to fortio.istio.io too.

@ldemailly what do you think about the availability concerns @rshriram has?

**Comment by ronenschafferibm** [comment]
The mesh-external traffic tests are enabled in Prow and CircleCI. They run and pass.
